### PR TITLE
Add Sentry!

### PR DESCRIPTION
**Pull request by ajbt200128**

### CODE REVIEW

File: src/main/kotlin/com/semgrep/idea/telemetry/SentryWrapper.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    private var appState: AppState = AppState.getInstance()
+    private fun enabled() = appState.lspSettings.metrics.enabled
+    private var initialized = false
+
+    fun init() {
+        if (enabled() && !initialized) {
+            Sentry.init { options ->
+                options.dsn = "https://4d1a49b9d3a6cd498af148cfdd479820@o77510.ingest.us.sentry.io/4507370251288576"
```
Comment (tpetr) [inline src/main/kotlin/com/semgrep/idea/telemetry/SentryWrapper.kt:31]: can we make this a const somewhere?

File: src/main/kotlin/com/semgrep/idea/telemetry/SentryWrapper.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        if (!initialized || !enabled()) {
+            return
+        }
+        val baseMap = mapOf(
+            "semgrepVersion" to appState.pluginState.semgrepVersion,
+            "isNewAppInstall" to appState.lspSettings.metrics.isNewAppInstall,
+            "extensionVersion" to appState.lspSettings.metrics.extensionVersion,
+            "loggedIn" to appState.pluginState.loggedIn,
+        )
+        val flatMap = appState.lspSettings.toFlatMap().filter { (t, _) ->
```
Comment (tpetr) [inline src/main/kotlin/com/semgrep/idea/telemetry/SentryWrapper.kt:60]: `flatMap` feels like not the right name for this variable

File: src/main/kotlin/com/semgrep/idea/telemetry/SentryProcessListener.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import java.time.format.DateTimeFormatter
+
+class SentryProcessListener : ProcessListener {
+
+    private var processLog = mutableListOf<String>()
+    private var sentry = SentryWrapper.getInstance()
+
+    private fun attachmentOfLog(): Attachment {
+        val humanReadableTimestamp = DateTimeFormatter.ISO_INSTANT.format(java.time.Instant.now())
+        return Attachment(processLog.joinToString("\n").toByteArray(), "lsp-output-$humanReadableTimestamp.log")
```
Comment (tpetr) [inline src/main/kotlin/com/semgrep/idea/telemetry/SentryProcessListener.kt:17]: similar to vscode, can we use an actual file instead?

File: src/main/kotlin/com/semgrep/idea/settings/LspSettings.kt [hunk 4]
```diff
@@ -23,6 +23,33 @@ data class SemgrepLspSettings(
     var heapSizeJS: Int = 4096,
 ) {
 
+    fun toFlatMap(): Map<String, String> {
```
Comment (tpetr) [inline src/main/kotlin/com/semgrep/idea/settings/LspSettings.kt:26]: `flatMap` has meaning in the JVM ecosystem -- `toFlattenedMap` might be more appropriate

File: src/main/kotlin/com/semgrep/idea/telemetry/SentryWrapper.kt (hunk 1 above)
Comment (ajbt200128) [reply src/main/kotlin/com/semgrep/idea/telemetry/SentryWrapper.kt:31]: yup

File: src/main/kotlin/com/semgrep/idea/telemetry/SentryProcessListener.kt (hunk 3 above)
Comment (ajbt200128) [reply src/main/kotlin/com/semgrep/idea/telemetry/SentryProcessListener.kt:17]: yea good catch

File: src/main/kotlin/com/semgrep/idea/lsp/SemgrepInstaller.kt [hunk 5]
```diff
@@ -52,15 +54,22 @@ object SemgrepInstaller {
...
             .build()
-        val response = client.send(request, HttpResponse.BodyHandlers.ofString()).body()
-        // Can't figure out how to actually parse this to a string, not a quoted string. Oh well
-        val version = Json.parseToJsonElement(response).jsonObject.get("latest").toString().replace("\"", "")
-        return SemVer.parseFromText(version)
+        try {
+            val response = client.send(request, HttpResponse.BodyHandlers.ofString()).body()
+            val versionObject = Json.parseToJsonElement(response).jsonObject
+            val latestVersion = versionObject["latest"].toString().replace("\"", "")
+            val minVersion = versionObject["versions"]!!.jsonObject["minimum"].toString().replace("\"", "")
```
Comment (tpetr) [inline src/main/kotlin/com/semgrep/idea/lsp/SemgrepInstaller.kt:68]: can we file a bug to track `// Can't figure out how to actually parse this to a string, not a quoted string. Oh well` ?

**Review by tpetr — APPROVED** [review]
LGTM if tested

**Pull request merged**